CONTRIBUTING.rst

```diff
@@ -48,54 +58,43 @@ ones. Use the ``celeritas-gen.py`` script (in the ``scripts/dev`` directory) to
-*constructive feedback* ("here's a suggestion to make this better" or "did you
-consider what would happen if X?"), not *destructive feedback* ("this code is
-ugly").
+Submitting code changes
+-----------------------
 
-Reviews should not be started until the "draft" status has been removed (if it
-was set to begin with). Once a pull request is under review, *do not* rebase,
-squash, or otherwise alter the branch history. It causes GitHub to lose
-comments and causes notifications to redirect. Additionally it will lose
-information about whether the test passed and failed on previous commits.
+When you believe that you've made a nontrivial, self-contained improvement to
+the code, it's time to create a `pull request`_ (PR) to get feedback on your
+changes before they're merged into the code base.
 
-The :ref:`code <code_guidelines>` and :ref:`style <style_guidelines>` guidelines
-must be followed for all new code and code changes.
-This includes the use of the correct formatting as well as the addition of
-documentation and unit tests for new code and bug fixes.
+Before opening the pull request, check that the :ref:`code <code_guidelines>`
+and :ref:`style <style_guidelines>` guidelines have been followed for all new
+code and code changes.  Ensure the use of the correct formatting as well as the
+addition of documentation and unit tests for new code and bug fixes.
 
 All tests must pass on the CI runner before a PR can be merged. (Exceptions
 will be made if any failures are clearly unrelated to the changes and enough
 tests and/or configuration are passing to show that the new code is working.
 For example, some of the configurations have a tendency to fail due to disk
-space issues.)
-
-Since there are few enough merge requests these days, only :ref:`maintainers
-<roles>` may commit a merge.
-
-
-Ownership and Authorship
-------------------------
-
-The person who writes a line of code is its author and copyright holder, but
-the new code should be the shared responsibility of the project rather than the
-exclusive property of a single contributor.
-Celeritas is a collaborative project with *collective* ownership.
-If the code is reviewed according to the guidelines above, at least two people
-should always be comfortable modifying any piece of code.
+space issues.) It's best to test locally first before submitting your pull
+request, and keep in mind that the multiple configurations on the CI (different
+dependency versions, different features) may reveal failures that your local
+testing might have missed.
+
+Each pull request should be assigned one or two reviewers who will provide
+constructive feedback on your changes. Their responsibilities are outlined in
+:ref:`code_review`.
+Reviews should not be started until the "draft" status has been removed (if you
+marked it as a draft initially). Once a pull request is under review, *do not*
+rebase, squash, or otherwise alter the branch history. It causes GitHub to lose
```

Comment:
> It causes GitHub to lose comments

For the record, the comments are never lost, just harder to find and connect to the code ;)  [Namely they are still in the discussion stream][And related if a further commit completely remove the code the comment was applying to, the same thing applies to the comment.  

Still not rewriting the history helps the reviewer easily see the deltas.

Reply:
I swear I've had discussion stream comments disappear before when force-pushing: and the notification links for sure point to "not found" pages.

Reply:
> and the notification links for sure point to "not found" pages.

Since the notification links point to the commit and the commit can/will be garbage collected, indeed this also what I see.

> I swear I've had discussion stream comments disappear before when force-pushing: 

In the context of ROOT, we always rebase and push to updates PR and I do not see the comments disappear.  Sometimes they are hard to find ... especially if they have been marked as resolved or if the conversation is so long that github decide to elide most of the conversation (with a link in the middle of the conversation 'click here for the old stuff' (paraphrasing :) )).

Reply:
```suggestion
rebase, squash, or otherwise alter the branch history.  It makes finding older
comments much harder and break notifications.  More importantly having 
new commits for the additions and new modifications makes it much easier
for the reviewer to keep track of the progress and changes. 
You *can* merge the main upstream branch if
```